pkg/prometheus/promcfg.go

```diff
@@ -4044,6 +4043,100 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		})
 	}
 
+	// ScaleWaySDConfig
+	if len(sc.Spec.ScaleWaySDConfigs) > 0 {
+		configs := make([][]yaml.MapItem, len(sc.Spec.ScaleWaySDConfigs))
+		for i, config := range sc.Spec.ScaleWaySDConfigs {
+			configs[i] = append(configs[i], yaml.MapItem{
+				Key:   "access_key",
+				Value: config.AccessKey,
+			})
+
+			value, _ := s.GetSecretKey(config.SecretKey)
+			configs[i] = append(configs[i], yaml.MapItem{
+				Key:   "secret_key",
+				Value: string(value),
+			})
+
+			configs[i] = append(configs[i], yaml.MapItem{
+				Key:   "project_id",
+				Value: config.ProjectID,
+			})
+
+			configs[i] = append(configs[i], yaml.MapItem{
+				Key:   "role",
+				Value: strings.ToLower(string(config.Role)),
+			})
+
+			if config.Port != nil {
+				configs[i] = append(configs[i], yaml.MapItem{
+					Key:   "port",
+					Value: config.Port,
+				})
+			}
+
+			if config.ApiURL != nil {
+				configs[i] = append(configs[i], yaml.MapItem{
+					Key:   "api_url",
+					Value: *config.ApiURL,
+				})
+			}
+
+			if config.Zone != nil {
+				configs[i] = append(configs[i], yaml.MapItem{
+					Key:   "zone",
+					Value: config.Zone,
+				})
+			}
+
+			if config.NameFilter != nil {
+				configs[i] = append(configs[i], yaml.MapItem{
+					Key:   "name_filter",
+					Value: config.NameFilter,
+				})
+			}
+
+			if len(config.TagsFilter) > 0 {
+				configs[i] = append(configs[i], yaml.MapItem{
+					Key:   "tags_filter",
+					Value: config.TagsFilter,
+				})
+			}
+
+			if config.RefreshInterval != nil {
+				configs[i] = append(configs[i], yaml.MapItem{
+					Key:   "refresh_interval",
+					Value: config.RefreshInterval,
+				})
+			}
+
+			configs[i] = cg.addProxyConfigtoYaml(configs[i], s, config.ProxyConfig)
+
+			if config.TLSConfig != nil {
```

Comment:
Could we do the same thing that we do with `ProxyConfig` etc,. and just handle the nil value of `TLSConfig` within the `addSafeTLSToYaml`? @simonpasquier @slashpai

Reply:
I can optimize it in the next PR.